qiskit_aer/backends/pulse_simulator.py

```diff
@@ -278,6 +278,11 @@ def _execute(self, qobj):
         qobj.config.qubit_freq_est = self.defaults().qubit_freq_est
         return pulse_controller(qobj)
 
+    def _execute_direct(self, circuits, noise_model, config):
+        """Execute circuits on the backend.
+        """
+        raise AerError("not implemented yet")
```

Comment:
Will this ever get called? `_execute_direct` isn't a required part of inheriting from `AerBackend` and nothing in the pulse path as defined in `AerBackend` should ever call this, so I'd say lets just remove this.

Reply:
`_execute_circuits` (renamed from `_execute_direct`) is defined as `@abstractmethod` in `AerBackend`. So, we need to implement it for `PulseSimulator` or take `@abstractmethod`  from `AerBackend`. But if we take `@abstractmethod`, lint [errors](https://github.com/Qiskit/qiskit-aer/actions/runs/4239315616/jobs/7367249914) call of `_execute_direct` because  it does not return any. Of cause `_execute_direct` can return `None` and this lint error will be worked around, but I think keeping `@abstractmethod` and implementing `_execute_circuits` in deprecating `PulseSimulator` is reasonable.

```
************* Module qiskit_aer.backends.aerbackend
qiskit_aer/backends/aerbackend.py:432:8: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```